### PR TITLE
wallet: don't include bdb files from our headers

### PR DESCRIPTION
**Pull request by theuni**
Only `#include` upstream bdb headers from our cpp files.

It's generally good practice to avoid including 3rd party deps in headers as otherwise they tend to sneak into new compilation units. IMO this makes for a nice cleanup.

There's a good bit of code movement here, but each commit is small and _should_ be obviously correct.

Note: in the future, the buildsystem can add the bdb include path for `bdb.cpp` and `salvage.cpp` only, rather than all wallet sources.

### CODE REVIEW

**Comment by achow101** [comment]
ACK 2d09d0f50408f9c522b9efa4f386072502c7b3d1

Changes are primarily moves and appear otherwise obviously correct.

**Review by maflcko — APPROVED** [review]
lgtm

**Review by hebasto — APPROVED** [review]
Approach ACK 2d09d0f50408f9c522b9efa4f386072502c7b3d1.

Suggesting to adjust our IWYU mapping file:
```diff
--- a/contrib/devtools/iwyu/bitcoin.core.imp
+++ b/contrib/devtools/iwyu/bitcoin.core.imp
@@ -4,4 +4,7 @@
   { include: [ "<bits/termios-struct.h>", private, "<termios.h>", public ] },
   { include: [ "<bits/termios-tcflow.h>", private, "<termios.h>", public ] },
   { include: [ "<bits/chrono.h>", private, "<chrono>", public ] },
+
+  # Berkeley DB
+  { include: [ "<db.h>", private, "<db_cxx.h>", public ] },
 ]
```

or `#include <db.h>` explicitly along with `#include <db_cxx.h>`.

File: src/wallet/bdb.cpp [hunk 1]
```diff
@@ -275,6 +271,11 @@ SafeDbt::operator Dbt*()
     return &m_dbt;
 }
 
+Span<const std::byte> SpanFromDbt(const SafeDbt& dbt)
```
[the remark this answers is not in the document]
Comment (theuni) [reply src/wallet/bdb.cpp]: Nice catch. Fixed.

**Review by hebasto — APPROVED** [review]
ACK 8b5397c00e821d7eaab22f512e9d71a1a0392ebf

**Comment by theuni** [comment]
@hebasto I'm ~0 on doing either/none of those changes. Happy to make a change if you or @achow101 or @MarcoFalke have a strong preference, otherwise I'll leave it as-is.

**Comment by achow101** [comment]
reACK 8b5397c00e821d7eaab22f512e9d71a1a0392ebf

**Pull request merged**